cli/command/container/completion.go

```diff
@@ -125,6 +126,23 @@ func completeIpc(cli command.Cli) func(cmd *cobra.Command, args []string, toComp
 	}
 }
 
+// completeLink implements shell completion for the `--link` option  of `run` and `create`.
+func completeLink(cli command.Cli) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return postfixWith(":", containerNames(cli, cmd, args, toComplete)), cobra.ShellCompDirectiveNoSpace
+	}
+}
+
+// containerNames contacts the API to get names and optionally IDs of containers.
+// In case of an error, an empty list is returned.
+func containerNames(dockerCLI completion.APIClientProvider, cmd *cobra.Command, args []string, toComplete string) []string {
+	names, _ := completion.ContainerNames(dockerCLI, true)(cmd, args, toComplete)
+	if names == nil {
+		return []string{}
+	}
```

Comment:
Just curious; is this a bug in `completion.ContainerNames` and should we update that one?

Reply:
`completion.ContainerNames` is consistent with the other helper functions in completion.go.
They all return `nil, cobra.ShellCompDirectiveError` in case of api errors, which is an appropriate way of signaling completion problems.

I just wanted to get a list of container names that I can feed into `postfixWith` directly without having to bother with api errors.
That's why I adapted the existing function to my use case.
I had to do so anyway because I had to get rid of the ShellCompDirective.

From my point of view, a better solution would have been to pull out a function from `completion.ContainerNames` that gets the container names directly as an array (no `nil`).
But this would have caused naming issues: This function should obviously be named `completion.ContainerNames` and thus collide with the existing function.
So the existing function, and for consistency similar functions, would all have to be renamed (`completion.ContainerNames` -> `completion.CompleteContainerNames`, `completion.NetworkNames` -> `completion.NetworkNames`, ...)